### ISSUE TITLE
Add stages to CodePipeline

### ISSUE THREAD

**Issue by rfc2119** [aws/copilot-cli#4552]
Hi, great tool you have there! I would like to add stages to my pipeline, however it seems there's no way currently to add a stage with its actions. I have other stages I would like to incorporate other than build and deploy. What I currently do is:
1. `copilot pipeline deploy` 
2. Grab the result template from CloudFormation 
3. Modify it by hand to add new stages
4. `aws cloudformation deploy`

I have searched the issues and so far there isn't one that describes this. Is there an existing ticket I am not aware of ? Thanks

**Comment by iamhopaul123** [comment]
Hello @rfc2119. Thank you for your appreciation! Could you elaborate what stages are you looking to add to the pipeline?  Would you mind sharing with us your modified CFN template specifically for the pipeline resource?

**Comment by rfc2119** [comment]
Hello @iamhopaul123. Thank you for your response. The stages depend on purely custom buildspecs and there is no standard, but we mainly integrate the DevSecOps cycle. For example, for repositories written in NodeJs, we run a basic static code analysis with `npm audit`. For Go projects we use `gosec`, and so on. We also lint and format the code.

I imagine this can be done by integrating a custom buildspec and assign it to a custom stage. Do you have any ideas ? Moreover, are you planning to go over this anytime soon ?

**Comment by iamhopaul123** [comment]
Understood. Do you think it'll help integrate the DevSecOps cycle if we support https://github.com/aws/copilot-cli/issues/3007? Or it has to be a separate stage?

**Comment by rfc2119** [comment]
No, code linting and DevSecOps stages are two stages that run before the deployment. We have also some custom steps that we perform within each stage.
